lacci/lib/shoes/widgets/para.rb

```diff
@@ -27,10 +27,18 @@ def initialize(*args, stroke: nil, size: :para, font: nil, hidden: false, **html
     def text_children_to_items(text_children)
       text_children.map { |arg| arg.is_a?(String) ? arg : arg.linkable_id }
     end
+  def replace(*children)
+      update_text_children(*children)
+    end
 
-    def replace(*children)
-      @text_children = children
+    def text=(*children)
```

Comment:
Will this ever be multiple items? I think an assignment will always just get one item. Not sure how this would work if you assigned an array.

Reply:
hm , ok n also  in replace does it make diff if we use splat operator or no?

Reply:
Looks like sometimes it does?

```
noah@Noahs-MBP-3 scarpe % irb
irb(main):001> class Bobo; def text=(val); @text=val; end; def tsplat=(*args); @tsplat = args; end; end
=> :tsplat=
irb(main):002> b = Bobo.new
=> #<Bobo:0x00000001045f0c30>
irb(main):003> class Bobo; attr_reader :text; attr_reader :tsplat; end
=> [:tsplat]
irb(main):004> b.text
=> nil
irb(main):006> b.text = *[1, 2, 3]
=> [1, 2, 3]
irb(main):007> b.text
=> [1, 2, 3]
irb(main):008> b.tsplat = *[1, 2, 3]
=> [1, 2, 3]
irb(main):009> b.tsplat
=> [[1, 2, 3]]
```

Reply:
hm okay, but why do we/user pass array  in either replace or in text...,
as of now even if we use splat operator if user passes anything other than string it just raises error if it dont have linkable id...

Reply:
Normally you wouldn't use a splat operator in an assignment. If the user wants to assign an array, they can just assign the array value. Something like:

```ruby
def text=(val)
  @text=val
end

# or more succinctly:

attr_writer(:text)
```

Reply:
Though never mind, attr_writer is no good because we need to make sure text gets sent via the text_children display property.